precompiles/regular/regular.go

```diff
@@ -0,0 +1,262 @@
+package regular
```

Comment:
Just for my understanding on this file.
Has it been fully written from scratch, written from a template, or generated then customized?

We should have a better unit test coverage on it (though we might make exception for this one as a test contract) then we should find some way to exclude test coverage from the generated/generic part but not the customized part

Reply:
This contract is part of bharvest work, I'm working currently to introduce unit testing and also switching from regularCall to a call that get the gas.

Reply:
So does it means this entire file is written from scratch?

Reply:
Correct, the precompiled contracts are written from scratch and mostly following the same structure that you can see in prototype.go